tensorzero-core/src/config/mod.rs

```diff
@@ -1251,6 +1420,69 @@ impl Config {
             })?
             .clone())
     }
+
+    /// Overlays runtime-specific configuration from a live config onto this config.
+    ///
+    /// When using a config loaded from a historical snapshot, certain fields should
+    /// reflect the current runtime environment rather than the snapshot's values:
+    /// - `gateway`: Runtime gateway settings (bind address, auth, observability, etc.)
+    /// - `object_store_info`: Current object store configuration
+    /// - `postgres`: Current postgres settings
+    /// - `rate_limiting`: Current rate limiting rules
+    /// - `http_client`: Current HTTP client configuration
+    ///
+    /// These fields are infrastructure/runtime concerns that should not be replayed
+    /// from historical data.
+    pub fn overlay_runtime_config(self, live_config: &Config) -> Config {
+        let gateway = GatewayConfig {
```

Comment:
Is this because `GatewayConfig` doesn't derive `Clone`? (why not?)

Reply:
nah it's just so we are very explicit about the fields and how we're getting them